elements/rh-pagination/rh-pagination.ts

```diff
@@ -0,0 +1,179 @@
+   * we can't do `::slotted(nav ol li)`, we need to reflect
+   * it to a host attribute, so that lightdom CSS can target
+   * the list items.
+   */
+  @property({ reflect: true }) overflow: 'start' | 'end' | 'both' | null = null;
+
+  /** Accessible label for the 'first page' button */
+  @property({ attribute: 'label-first' }) labelFirst = 'first page';
+  /** Accessible label for the 'previous page' button */
+  @property({ attribute: 'label-previous' }) labelPrevious = 'previous page';
+  /** Accessible label for the 'next page' button */
+  @property({ attribute: 'label-next' }) labelNext = 'next page';
+  /** Accessible label for the 'last page' button */
+  @property({ attribute: 'label-last' }) labelLast = 'last page';
+
+  #screenSize = new ScreenSizeController(this);
+  #logger = new Logger(this);
+
+  #nav = this.querySelector('nav');
+  #links = this.#nav?.querySelectorAll<HTMLAnchorElement>('li a');
+  #currentLink = this.#getCurrentLink();
+
+  #currentIndex = -1;
+  #firstLink: HTMLAnchorElement | null = null;
+  #lastLink: HTMLAnchorElement | null = null;
+  #nextLink: HTMLAnchorElement | null = null;
+  #prevLink: HTMLAnchorElement | null = null;
+
+  update(changed: PropertyValues<this>): void {
+    this.#update();
+    super.update(changed);
+  }
+
+  render() {
+    const { mobile, size } = this.#screenSize;
+    const { labelFirst, labelPrevious, labelNext, labelLast } = this;
+    const firstHref = this.#currentLink === this.#firstLink ? undefined : this.#firstLink?.href;
+    const prevHref = this.#prevLink?.href;
+    const nextHref = this.#nextLink?.href;
+    const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
+    return html`
+      <div id="container" class=${classMap({ mobile, [size as string]: true })}>
+        <a id="first" class="stepper" href=${ifDefined(firstHref)} aria-label=${labelFirst}>${L2}</a>
+        <a id="prev" class="stepper" href=${ifDefined(prevHref)} aria-label=${labelPrevious}>${L1}</a>
+
+        <div id="nav-container" ?hidden=${mobile}>
+          <slot @slotchange=${this.#update}></slot>
+        </div>
+
+        <a id="next" class="stepper" href=${ifDefined(nextHref)} aria-label=${labelNext}>${L1}</a>
+        <a id="last" class="stepper" href=${ifDefined(lastHref)} aria-label=${labelLast}>${L2}</a>
+
+        <div id="numeric" ?hidden=${!mobile} ?inert=${!mobile}>
+          <span id="go-to-page">
+            <slot name="go-to-page">Go to page</slot>
+          </span>
+          <input inputmode="numeric" aria-labelledby="go-to-page" value=${(this.#currentIndex ?? 0) + 1} />
+          <slot name="out-of">of</slot>
+          <a href=${ifDefined(lastHref)}>${this.#links?.length}</a>
+        </div>
+      </div>
+    `;
+  }
+
+  #update() {
+    this.#updateLightDOMRefs();
+    this.overflow = this.#getOverflow();
+    this.#validateA11y();
+  }
+
+  #getOverflow(): 'start' | 'end' | 'both' | null {
+    const overflowAt = 9;
```

Comment:
Should we make this an API option?

Reply:
any less than 9 doesn't really work for us. we could maybe make larger overflows optional, but design didn't ask for it

Reply:
Oh gotcha.  That makes sense.